packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -87,106 +69,70 @@ export const createOrEditPractitioners = (
-    serve
-      .update(keycloakUserValues)
-      .then(() => {
-        createOrEditPractitioners(
-          opensrpBaseURL,
-          opensrpServiceClass,
-          {
-            ...values,
-            userId,
-          },
-          practitioner,
-          isEditing
-        );
-        setSubmitting(false);
-        sendSuccessNotification(MESSAGE_USER_EDITED);
-        history.push(URL_USER);
-      })
-      .catch((_: Error) => {
-        setSubmitting(false);
-        sendErrorNotification(ERROR_OCCURED);
-      });
+  delete keycloakUserValue.active;
+  delete keycloakUserValue.userGroup;
+  delete keycloakUserValue.practitioner;
+
+  if (values.id) {
+    const serve = new KeycloakService(`${KEYCLOAK_URL_USERS}/${values.id}`, keycloakBaseURL);
+    await serve.update(keycloakUserValue);
   } else {
-    const serve = new keycloakServiceClass(KEYCLOAK_URL_USERS, keycloakBaseURL);
-    serve
-      .create({
-        ...keycloakUserValues,
-        enabled: true,
-      })
-      .then((response: Response | undefined) => {
-        // workaround to get userId for newly created user
-        // immediately after performing a POST
-        const newValues = response ? buildUserObject(response, values) : values;
-        createOrEditPractitioners(
-          opensrpBaseURL,
-          opensrpServiceClass,
-          newValues,
-          practitioner,
-          isEditing
-        );
-        setSubmitting(false);
-        sendSuccessNotification(MESSAGE_USER_CREATED);
-      })
-      .catch((_: Error) => {
-        setSubmitting(false);
-        sendErrorNotification(ERROR_OCCURED);
-      });
+    const serve = new KeycloakService(KEYCLOAK_URL_USERS, keycloakBaseURL);
+    const response: Response | undefined = await serve.create({
+      ...keycloakUserValue,
+      enabled: true,
+    });
+    // workaround to get user Id for newly created user immediately after performing a POST
+    values = response ? buildUserObject(response, values) : values;
   }
-};
 
-/** interface user action */
-export interface UserAction {
-  alias: string;
-  name: string;
-  providerId: string;
-  enabled: boolean;
-  defaultAction: boolean;
-  priority: number;
-  config: Dictionary;
-}
+  await createOrEditPractitioners(opensrpBaseURL, values);
+
+  // Assign User Group to user
+  const promises: Promise<void>[] = [];
+
+  values.userGroup?.forEach((groupId) => {
+    const userGroupValue = userGroups.find((group) => group.id === groupId) as UserGroup;
+
+    const serve = new KeycloakService(
+      `${KEYCLOAK_URL_USERS}/${values.id}${KEYCLOAK_URL_USER_GROUPS}/${groupId}`,
+      keycloakBaseURL
+    );
+
+    const promise = serve.update(userGroupValue);
+    promises.push(promise);
+  });
+
+  await Promise.allSettled(promises).catch((_: Error) => sendErrorNotification(ERROR_OCCURED));
+  sendSuccessNotification(MESSAGE_USER_GROUP_EDITED);
+
+  sendSuccessNotification(MESSAGE_USER_EDITED);
```

Comment:
should these  2 statements be called within the then block of a promise
? if the promise runs into an error, do we still want these 2 to be called

Reply:
if any of the promise fails it will throw the error and the code won't even reach this loc

if it reaches this loc it will show two success one for the location group update and one for overall. i added this groups success one cuz we were showing one for practitioner update too ..

i think we should only show one notification instead of showing all 3